### PR TITLE
💸 Dynamic service rates

### PR DESCRIPTION
**Pull request by M4tini**
https://myparcelcombv.atlassian.net/browse/MP-5200
---
- Added `is_dynamic` to ServiceRate resource.
- Added `resolveDynamicServiceRates()` function that communicates with our `/get-dynamic-service-rates` RPC endpoint.
- Updated `getServiceRatesForShipment()` and the `PriceCalculator` to recognize dynamic rates and resolve their price.

TODO
- Write tests

### CODE REVIEW

File: src/Resources/ServiceRate.php [hunk 1]
```diff
@@ -353,4 +359,32 @@ public function getServiceOptions()
...
     }
+
+    public function setIsDynamic($isDynamic)
+    {
+        $this->attributes[self::ATTRIBUTE_IS_DYNAMIC] = $isDynamic;
+
+        return $this;
+    }
+
+    public function getIsDynamic()
```
Comment (NickVries) [inline src/Resources/ServiceRate.php:370]: `isDynamic` reads just a little bit better here :)

```suggestion
    public function isDynamic()
```

File: src/MyParcelComApi.php [hunk 2]
```diff
@@ -398,6 +417,53 @@ public function getServiceRatesForShipment(ShipmentInterface $shipment, $ttl = s
...
+                ],
+            ];
+        }
+
+        if (!isset($data['relationships']['service'])) {
+            throw new InvalidResourceException('Cannot retrieve dynamic rates without a service');
+        }
+        if (!isset($data['relationships']['contract'])) {
+            throw new InvalidResourceException('Cannot retrieve dynamic rates without a contract');
+        }
```
Comment (NickVries) [inline src/MyParcelComApi.php:459]: We have more requirements for this endpoint, like the sender and recipient addresses and weight. 
Would you happen to know how the SDK handles spec errors? It feels arbitrary to check service and contract, but not the other requirements here.

**Review by NickVries — CHANGES_REQUESTED** [review]
I'm missing some tests for the dynamic service rates feature.

File: src/MyParcelComApi.php (hunk 2 above)
Comment (M4tini) [reply src/MyParcelComApi.php]: I will 🔥 this because I have just added request validation on this endpoint in our API.

**Review by NickVries — APPROVED** [review]

**Pull request merged**